rust/noosphere-cli/src/native/workspace.rs

```diff
@@ -164,6 +167,13 @@ impl Workspace {
         Ok(())
     }
 
+    /// Use the provided `storage_config` when opening the [SphereContext].
+    /// Use [Workspace::release_sphere_context] after setting storage config
+    /// to ensure sphere is opened with these settings.
+    pub fn with_storage_config(&mut self, storage_config: StorageConfig) {
```

Comment:
What is the use case for configuring this lazily e.g., after the `Workspace` has been constructed?

Reply:
due to `invoke_cli(Cli, Workspace)` (or the change in this PR, `invoke_cli(Cli, &mut Workspace)`) -- though we could have `invoke_cli(Cli)` and `invoke_cli_with_workspace(Cli, &mut Workspace)` for scenarios where the workspace needs to be provided, and then this becomes a construction-time config